sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/OffHeapColumnVector.java

```diff
@@ -84,9 +84,7 @@ public long valuesNativeAddress() {
     return data;
   }
 
-  @Override
-  public void close() {
-    super.close();
+  private void releaseMemory() {
```

Comment:
nit: if we can move the declartion of this function to the parent class, then `reset` and `close` can both be implementd in the parent class. The child classes only need to implement `releaseMemory`.